lib/vagrant-libvirt/action/set_boot_order.rb

```diff
@@ -86,9 +86,13 @@ def final_boot_order(boot_order, devices)
 
         def search_network(nets, xml)
           str = '/domain/devices/interface'
-          str += "[(@type='network' or @type='udp' or @type='bridge')"
+          str += "[(@type='network' or @type='udp' or @type='bridge' or @type='direct')"
           unless nets.empty?
-            str += " and source[@network='#{nets.first['network']}']"
+            net = nets.first
+            network = net['network']
+            dev = net['dev']
+            str += " and source[@network='#{network}']" if network
+            str += " and source[@dev='#{dev}']" if dev
```

Comment:
Think add a note to the readme to indicate that the boot_network can contain a `dev` reference as well.

For my better understanding, are these exclusive options? as in can the source contain both? or is it going to be only one or the other? It looks like they cannot both be set based on https://libvirt.org/formatdomain.html#network-interfaces

Reply:
I've added an example for the `public_network` that uses the `dev` filter.

I also think they cannot be used together, but I'm not really sure (I could not find a formal libvirt xml schema that describes all the available options to confirm; the current docs are somewhat dense to read and understand).

Do you want me to raise an error when we try to use both `network` and `dev` in the filter?

Reply:
I don't think so, not yet anyway, when we get to adding tests for more of the networking stack then it's possibly something that can be looked at when there is confidence on what XML blocks are actually acceptable.